contracts/mainnet/connectors/compound-import/main.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+import { TokenInterface, AccountInterface } from "../../common/interfaces.sol";
+import { CompoundHelper } from "./helpers.sol";
+import { Events } from "./events.sol";
+
+// 1. Get info for all the assets the user has supplied as collateral and the assets he borrowed.
+// 2. Using the flash loan funds, pay back the user's debt in the EOA account.
+// 3. After paying the debt, transfer the user's tokens from EOA to DSA.
+// 4. Then borrow debt of same tokens but include flash loan fee in it.
+
+contract CompoundImportResolver is CompoundHelper {
+
+    /**
+     * @notice this function performs the import of user's Compound positions into its DSA
+     * @dev called internally by the importCompound and migrateCompound functions
+     * @param _importInputData the struct containing borrowIds of the users borrowed tokens
+     * @param _flashLoanFee flash loan fee
+     */
+    function _importCompound(
+        ImportInputData memory _importInputData,
+        uint256 _flashLoanFee
+    ) internal returns (string memory _eventName, bytes memory _eventParam) {
+        require(AccountInterface(address(this)).isAuth(_importInputData.userAccount), "user-account-not-auth");
+
+        require(_importInputData.supplyIds.length > 0, "0-length-not-allowed");
+
+        ImportData memory data;
+
+        uint _length = add(_importInputData.supplyIds.length, _importInputData.borrowIds.length);
+        data.cTokens = new address[](_length);
+
+        // get info about all borrowings and lendings by the user on Compound
+        data = getBorrowAmounts(_importInputData, data);
+        data = getSupplyAmounts(_importInputData, data);
+
+        for(uint i = 0; i < data.cTokens.length; i++){
+            enterMarket(data.cTokens[i]);
+        }
```

Comment:
@pradyuman-verma @chinmay0402 I think it's better to get all the markets entered by dsa first. And then do the checks. Instead of fetching all the markets for dsa for every ctoken.